### PR TITLE
Preview/fe 1589 generate wallet deposit endpoint

### PR DESCRIPTION
**Pull request by sahma19**
Preview environment for generating Wallet Deposit Endpoint

### CODE REVIEW

File: libs/eo/transfers/src/lib/eo-transfers.service.ts [hunk 1]
```diff
@@ -86,4 +90,11 @@ export class EoTransfersService {
       `${this.#apiBase}/history/transfer-agreements/${transferAgreementId}`
     );
   }
+
+  createWalletDepositEndpoint() {
+    return this.http.post<EoWalletDepositEndpointResponse>(
+      'https://demo.energioprindelse.dk/api/transfer-agreements/wallet-deposit-endpoint',
```
Comment (CodeReaper) [inline libs/eo/transfers/src/lib/eo-transfers.service.ts:96]:
```suggestion
      `${this.#apiBase}/transfer-agreements/wallet-deposit-endpoint`,
```

Comment (sahma19) [reply libs/eo/transfers/src/lib/eo-transfers.service.ts]: Changed

File: libs/eo/transfers/src/lib/eo-transfers-wallet-modal-component.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          Close
+        </watt-button>
+      </watt-modal-actions>
+    </watt-modal>
+
+    <ng-template #error>
+      <watt-empty-state
+        icon="power"
+        title="An unexpected error occured"
+        message="We're sorry, we couldn't generate your key. Please try again later."
```
Comment (CodeReaper) [inline libs/eo/transfers/src/lib/eo-transfers-wallet-modal-component.ts:87]:
```suggestion
        message="We are sorry, we could not generate your key. Please try again later."
```

File: libs/ui-watt/src/lib/components/modal/watt-modal.component.html [hunk 3]
```diff
@@ -35,6 +35,9 @@ <h3 class="watt-modal-title">{{ title }}</h3>
   <ng-content select="watt-modal-actions"></ng-content>
 
   <div *ngIf="loading" class="watt-modal__spinner">
-    <watt-spinner></watt-spinner>
+    <div>
+      <watt-spinner></watt-spinner>
+      <p>{{ loadingMessage }} {{ loading }}</p>
```
Comment (CodeReaper) [inline libs/ui-watt/src/lib/components/modal/watt-modal.component.html:40]:
```suggestion
      <p>{{ loadingMessage }}</p>
```

File: libs/ui-watt/src/lib/components/modal/watt-modal.component.ts [hunk 4]
```diff
@@ -80,6 +80,10 @@ export class WattModalComponent implements AfterViewInit {
   @Input()
   loading = false;
 
+  /** Whether the modal should show a loading text for the loading state. */
+  @Input()
+  loadingMessage = '';
```
Comment (CodeReaper) [inline libs/ui-watt/src/lib/components/modal/watt-modal.component.ts:85]: Are there no tests associated with this component? (They should properly be expanded to have a few tests for this loading message)

File: libs/ui-watt/src/lib/components/modal/watt-modal.component.html (hunk 3 above)
Comment (sahma19) [reply libs/ui-watt/src/lib/components/modal/watt-modal.component.html:40]: Does not seem to break anything. so i am removing the last loading variable

**Review by CodeReaper — APPROVED** [review]

**Pull request merged**